android/src/main/kotlin/com/onegini/mobile/sdk/flutter/useCases/AuthenticateUserImplicitlyUseCase.kt

```diff
@@ -12,15 +12,18 @@ import io.flutter.plugin.common.MethodChannel
 
 class AuthenticateUserImplicitlyUseCase(private var oneginiClient: OneginiClient) {
     operator fun invoke(call: MethodCall, result: MethodChannel.Result) {
-        val scope = call.argument<ArrayList<String>>("scope")
-        val userProfile = oneginiClient.userClient.userProfiles.firstOrNull()
+        val profileId = call.argument<String>("profileId")
+        val scopes = call.argument<ArrayList<String>>("scopes")
+
+        val userProfile = oneginiClient.userClient.userProfiles.find { it.profileId == profileId }
+
         if (userProfile == null) {
             SdkError(USER_PROFILE_IS_NULL).flutterError(result)
             return
         }
-        oneginiClient.userClient.authenticateUserImplicitly(userProfile, scope?.toArray(arrayOfNulls<String>(scope.size)), object : OneginiImplicitAuthenticationHandler {
+        oneginiClient.userClient.authenticateUserImplicitly(userProfile, scopes?.toArray(arrayOfNulls<String>(scopes.size)), object : OneginiImplicitAuthenticationHandler {
```

Comment:
Whats going on here with this `arrayOfNulls`? 

I think we should just pass the scopes to the SDK instead of passing an array of null values to the sdk.

Reply:
done